### ISSUE TITLE
Emacs Flymake: diagnostics with empty span do not appear

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1104]
Example (assuming #1094 lands):
```javascript
switch (x) {
  case 1:
    f();
  case 2:
//^ warning should appear here, but doesn't
}
```
